### ISSUE TITLE
Allow querying by multiple citations through the db_rest_api

### ISSUE THREAD

**Issue by cthoyt** [sorgerlab/indra#602]
It would be very useful to be able to supply multiple citations in one query instead of having to make repeated calls to the API (and underlying database) 

https://github.com/sorgerlab/indra/blob/8ea86ae197ada0382c6ea0da3ad6c1aa466185d7/db_rest_api/api.py#L263

I could imagine changing the endpoint to take in a JSON object that has a list of identifiers in the same format, and slightly rearranging the call to `get_statements_by_paper` to take a pre-defined list of `trids` that can be handled in the `get_paper_statements` route function.

I would first break the `get_statements_by_paper` function up to have to delegate to a "`_get_statements_by_trid_list`" function, which could be more general to handling lists generated from single paper queries (as the call is now) or multiple paper queries (as I would like to suggest)

```python
def _get_statements_by_trid_list(trid_list, count=1000, db=None,
                                 do_stmt_count=False, preassembled=True):
    """Get the statements from a list of text reference identifiers.

    Parameters
    ----------
    trid_list : list[int]
        A list of text reference identifiers.
    count : int
        Number of statements to retrieve in each batch (passed to
        :py:func:`get_statements`).
    db : :py:class:`DatabaseManager`
        Optionally specify a database manager that attaches to something
        besides the primary database, for example a local databse instance.
    do_stmt_count : bool
        Whether or not to perform an initial statement counting step to give
        more meaningful progress messages.
    preassembled : bool
        If True, statements will be selected from the table of pre-assembled
        statements. Otherwise, they will be selected from the raw statements.
        Default is True.

    Returns
    -------
    A list of Statements from the database corresponding to the paper id given.
    """
    stmts = []
    for trid in trid_list:
        clauses = db.join(db.TextContent, db.RawStatements) \
                  + [db.TextContent.text_ref_id == trid]
        if preassembled:
            clauses += db.join(db.RawStatements, db.PAStatements)
        stmts.extend(get_statements(clauses, count=count, db=db,
                                    preassembled=preassembled,
                                    do_stmt_count=do_stmt_count))
    return stmts
```

Then, a new (and similar function) could be implemented to handle a list of `id_val` and `id_type` pairs:

```python
def get_statements_by_papers(id_val_type_pair_list, count=1000, db=None,
                            do_stmt_count=False, preassembled=True):
    """Get the statements from a list of papers.

    Parameters
    ----------
    id_val_type_pair_list : list[tuple[int or str, str]]
        A list of pairs of paper identifiers and types
        The value of the id for the paper whose statements you wish to retrieve.
    count : int
        Number of statements to retrieve in each batch (passed to
        :py:func:`get_statements`).
    db : :py:class:`DatabaseManager`
        Optionally specify a database manager that attaches to something
        besides the primary database, for example a local databse instance.
    do_stmt_count : bool
        Whether or not to perform an initial statement counting step to give
        more meaningful progress messages.
    preassembled : bool
        If True, statements will be selected from the table of pre-assembled
        statements. Otherwise, they will be selected from the raw statements.
        Default is True.

    Returns
    -------
    A list of Statements from the database corresponding to the paper id given.
    """
    if db is None:
        db = get_primary_db()

    trid_list = []

    for id_val, id_type in id_val_type_pair_list:
        this_trid_list = _get_trids(db, id_val, id_type)
        if not this_trid_list:
            continue
        trid_list.extend(this_trid_list)

    if not trid_list:
        return None

    return _get_statements_by_trid_list(trid_list, count=count,
                                        db=db, do_stmt_count=do_stmt_count,
                                        preassembled=preassembled)
```

It would even then be possible to reimplement `get_statements_by_papers` with a call to this more general function

```python
def get_statements_by_paper(id_val, id_type, count=1000, db=None,
                            do_stmt_count=False, preassembled=True):
    """Get the statements from a particular paper.

    Parameters
    ----------
    id_val : int or str
        The value of the id for the paper whose statements you wish to retrieve.
    id_type : str
        The type of id used (default is pmid). Options include pmid, pmcid, doi,
        pii, url, or manuscript_id. Note that pmid is generally the best means
        of getting a paper.
    count : int
        Number of statements to retrieve in each batch (passed to
        :py:func:`get_statements`).
    db : :py:class:`DatabaseManager`
        Optionally specify a database manager that attaches to something
        besides the primary database, for example a local databse instance.
    do_stmt_count : bool
        Whether or not to perform an initial statement counting step to give
        more meaningful progress messages.
    preassembled : bool
        If True, statements will be selected from the table of pre-assembled
        statements. Otherwise, they will be selected from the raw statements.
        Default is True.

    Returns
    -------
    A list of Statements from the database corresponding to the paper id given.
    """
    return get_statements_by_paper([id_val, id_type], count=count, db=db,
                                do_stmt_count=do_stmt_count, preassembled=preassembled)
```

**Comment by pagreene** [comment]
This was in a very early (proof of concept) stage when I wrote it. It has since been put on hold until the database infrastructure is updated so that it has the latest content, and more importantly so that preassembled statements can actually be linked back to the refs. This will open the door for a lot of exciting new features, which should start rolling out soon.

What you suggested is more or less where we imagined this feature going in the future.

We could definitely support getting statements from multiple papers, certainly on the client side. The web service has fairly strict time and memory limits, so it is likely this would need to be implemented with multiple queries to the web service.

**Comment by cthoyt** [comment]
That's great! I already mentioned to John and Ben, but I'd love to skype with you all sometime this or next week so I can explain exactly how I've been using the db_rest_api and client to support one of our curation projects. I will share all of the code I've been using as well!

**Comment by bgyori** [comment]
Multiple paper ID queries were implemented in #689
